src/backend/gl/src/tex.rs

```diff
@@ -1085,3 +1195,38 @@ pub fn generate_mipmap(gl: &gl::Gl, name: Texture, target: gl::types::GLenum) {
     gl.BindTexture(target, name);
     gl.GenerateMipmap(target);
 }}
+
+trait CompressionAware {
+    fn is_compressed(&self) -> bool;
+}
+
+impl CompressionAware for core::format::SurfaceType {
```

Comment:
this logic should probably be in `gfx_core` crate since it's not GL specific

Reply:
Yeah that probably will make sense when the formats are supported in more backends. Right now this is really just an implementation detail of the gl backend, not part of the public API. I'm not sure if it even _will_ be useful in other backends, so it makes sense to leave it private for now (it's not exactly complicated either).